releasenotes/notes/optimizer-minimize-5a5a1e9d67db441a.yaml

```diff
@@ -0,0 +1,46 @@
+---
+features:
+  - |
+    Add a `minimize` methods to all optimizers;
+    :class:`~qiskit.algorithms.optimizers.Optimizer` and derived classes.
+    This method mimics the signature of SciPy's `minimize` function and returns an
+    :class:`~qiskit.algorithms.optimizers.OptimizerResult`.
+
+    For example
+
+    .. code-block::python
+
+        import numpy as np
+        from qiskit.algorithms.optimizers import COBYLA
+
+        def loss(x):
+            return -(x[0] - 1) ** 2 - (x[1] + 1) ** 3
+
+        initial_point = np.array([0, 0])
+        optimizer = COBYLA()
+        result = optimizer.minimize(loss, initial_point)
+
+        optimal_parameters = result.x
+        minimum_value = result.fun
+        num_function_evals = result.nfev
+deprecations:
+  - |
+    Deprecate the `optimize` method for all the optimizers
+    (:class:`~qiskit.algorithms.optimizers.Optimizer` and derived classes). Instead
+    the `minimize` method should be used which mimics the signature of SciPy's `minimize`
+    function.
```

Comment:
Normally I like to put what deprecated means (it will be removed in a future release) but we can tweak the wording at release time